src-srv/utils/hocuspocus.ts

```diff
@@ -0,0 +1,64 @@
+import type { Hocuspocus } from '@hocuspocus/server'
+import { Server } from '@hocuspocus/server'
+import { Logger } from '@hocuspocus/extension-logger'
+import { Redis } from '@hocuspocus/extension-redis'
+import { Database } from '@hocuspocus/extension-database'
+import type { RedisCache } from './RedisCache.ts'
+import type { Repository } from './Repository.ts'
+import { decodeJwt } from 'jose'
+import { initDoc } from './transformations/index.ts'
+
+interface CreateServerOptions {
+  repository: Repository
+  cache: RedisCache
+}
+
+export async function createServer(options: CreateServerOptions): Promise<Hocuspocus> {
+  const { repository, cache } = options
+
+  const server = Server.configure({
+    port: parseInt(process.env.API_PORT),
+    extensions: [
+      new Logger(),
+      new Redis({
+        host: process.env.REDIS_HOST, // FIXME: Should not use env
+        port: parseInt(process.env.REDIS_PORT) // FIXME: SHould not use env
```

Comment:
Should it use the cache? Not sure how it works, if you can use the same connection for persisting of docs and pub/sub subscription?

Reply:
I think it is needed. It might be that the code for the Database extension is wrong. It should probably not fetch from redis as it is doing now. Let's investigate it together

Reply:
For documentation: I think this is needed with multiple instances, not needed if the instance already have a cached version of the document. But if we're running multiple instances and the current instance haven't served it previously we 
need to fetch the complete document from redis. 

But there's probably room for improvement.